packages/devtools/lib/src/ui/viewport_canvas.dart

```diff
@@ -167,47 +173,14 @@ class ViewportCanvas extends Object with SetStateMixin {
       ..overflow = 'hidden';
     _element.add(_content);
 
-    final spacers = <CoreElement>[
-      _spaceTop,
-      _spaceBottom,
-      _spaceLeft,
-      _spaceRight
-    ];
-    for (var spacer in spacers) {
-      spacer.element.style
-        ..position = 'absolute'
-        ..overflow = 'hidden';
-    }
-    _spaceLeft.element.style
-      ..top = '0'
-      ..bottom = '0'
-      ..left = '0'
-      ..width = '0';
-    _spaceRight.element.style
-      ..top = '0'
-      ..bottom = '0'
-      ..right = '0'
-      ..width = '0';
-    _spaceTop.element.style
-      ..top = '0'
-      ..left = '0'
-      ..right = '0'
-      ..height = '0';
-    _spaceBottom.element.style
-      ..bottom = '0'
-      ..left = '0'
-      ..right = '0'
-      ..height = '0';
-
-    _content.add(spacers);
-
-    // TODO(jacobr): remove call to allowInterop once
-    // https://github.com/dart-lang/sdk/issues/35484 is fixed.
-    _visibilityObserver = IntersectionObserver(
-        allowInterop(_onVisibilityChange), {'root': _element.element});
-    for (var spacer in spacers) {
-      _visibilityObserver.observe(spacer.element);
-    }
+    // TODO(jacobr): clean this code up when
+    // https://github.com/dart-lang/html/issues/104 is fixed.
+    final observer = ResizeObserver(allowInterop((List<dynamic> entries, _) {
+      // XXX _scheduleRebuild();
```

Comment:
nit: can we remove this comment?

Reply:
yes. Switched to that code. Any time you see XXX in my code it means I missed fixing something before I send out for review.